### PR TITLE
Cleanup: use one expression to read usize from env var

### PR DESCRIPTION
**Pull request by mwtian**
This addresses the comment at https://github.com/MystenLabs/mysten-infra/pull/118#discussion_r943921496

### CODE REVIEW

File: crates/typed-store/src/rocks/mod.rs [hunk 1]
```diff
@@ -438,22 +439,17 @@ where
...
+        .tap_err(|e| info!("Env var {} is not set: {}", var_name, e))
+        .ok()?
+        .parse::<usize>()
+        .tap_err(|e| {
+            info!(
+                "Env var {} does not contain valid usize integer: {}",
+                var_name, e
+            )
+        })
+        .ok()
```
Comment (huitseeker) [inline crates/typed-store/src/rocks/mod.rs:452]:
```suggestion
    env::var(var_name).and_then(|v|.parse::<usize>()
        .tap_err(|e| {
            info!(
                "Env var {} unset or does not contain valid usize integer: {}",
                var_name, e
            )
        }))
        .ok()
```

**Review by huitseeker — APPROVED** [review]
Thanks!

File: crates/typed-store/src/rocks/mod.rs (hunk 1 above)
Comment (mwtian) [reply crates/typed-store/src/rocks/mod.rs:452]: It seems the signature of `Result<U, E>::and_then()` is 
```
pub fn and_then<U, F: FnOnce(T) -> Result<U, E>>(self, op: F) -> Result<U, E>
```
So the `Result` and the closure in `and_then()` must have the same error type. Here `env::var()` and `parse::<usize>()` have different error types.

Comment (huitseeker) [reply crates/typed-store/src/rocks/mod.rs:452]: Yup, I'd expect a `map_err` to come in at some point. I trust you to find the best compromise.

Comment (mwtian) [reply crates/typed-store/src/rocks/mod.rs:452]: Sounds good.

**Pull request merged**